egs/wsj/s5/steps/nnet3/make_unfoldedlstm_configs.py

```diff
@@ -0,0 +1,421 @@
+                use_presoftmax_prior_scale,
+                final_layer_normalize_target,
+                ng_per_element_scale_options, ng_affine_options,
+                label_delay, include_log_softmax, add_final_sigmoid,
+                self_repair_scale,
+                objective_type):
+
+    parsed_splice_output = ParseSpliceString(splice_indexes_string.strip(), label_delay)
+
+    left_context = parsed_splice_output['left_context']
+    right_context = parsed_splice_output['right_context']
+    num_hidden_layers = parsed_splice_output['num_hidden_layers']
+    splice_indexes = parsed_splice_output['splice_indexes']
+
+    if (num_hidden_layers < num_lstm_layers):
+        raise Exception("num-lstm-layers : number of lstm layers has to be greater than number of layers, decided based on splice-indexes")
+
+    prior_scale_file = '{0}/presoftmax_prior_scale.vec'.format(config_dir)
+
+    config_lines = {'components':[], 'component-nodes':[]}
+
+    config_files={}
+    assert(lstm_delay[0][0] < 0)
+    prev_layer_output = nodes.AddInputLayer(config_lines, feat_dim, splice_indexes[0], ivector_dim)
+ 
+    # Add the init config lines for estimating the preconditioning matrices
+    init_config_lines = copy.deepcopy(config_lines)
+    init_config_lines['components'].insert(0, '# Config file for initializing neural network prior to')
+    init_config_lines['components'].insert(0, '# preconditioning matrix computation')
+    nodes.AddOutputLayer(init_config_lines, prev_layer_output)
+    config_files[config_dir + '/init.config'] = init_config_lines
+
+    if add_lda:
+        prev_layer_output = nodes.AddLdaLayer(config_lines, "L0", prev_layer_output, config_dir + '/lda.mat')
+        # we moved the first splice layer to before the LDA..
+        # so the input to the first affine layer is going to [0] index
+        splice_indexes[0] = [0]
+
+    for i in range(num_lstm_layers):
+        if len(lstm_delay[i]) == 2: # bidirectional LSTM case, add both forward and backward 
+            prev_layer_output_forward = nodes.AddLstmLayer(config_lines,
+                                            "BUnfoldedLstm{0}_forward".format(i+1),
+                                            prev_layer_output, cell_dim,
+                                            recurrent_projection_dim, non_recurrent_projection_dim,
+                                            clipping_threshold, norm_based_clipping,
+                                            ng_per_element_scale_options = ng_per_element_scale_options,
+                                            ng_affine_options = ng_affine_options,
+                                            lstm_delay = lstm_delay[i][0],
+                                            self_repair_scale = self_repair_scale)
+            prev_layer_output_backward = nodes.AddLstmLayer(config_lines,
+                                            "BUnfoldedLstm{0}_backward".format(i+1),
```

Comment:
Please ignore this script right now since I have not changed the unfolded LSTM script as in unfolded RNN, and check AddUnfoldedRnnLayer() in components.py and make_unfoldedrnn_configs.py